Add new RHS trim function

rezamini opened the pull request:
This Pull Request introduces a new RHS "trim" function that takes a single string symbol argument and returns the same string with leading and trailing whitespace removed. 

Similar to PR #137 , this function is based on the specifications outlined in the Soar Manual documentation (3.3.6.6 Right-hand side Functions, page 80). The proposed changes have undergone testing, and a test file has been included to ensure the accurate implementation of the new RHS trim function.

rezamini committed:
- added trim RHS function that takes a single...
string symbol argument and returns the same string with leading and trailing whitespace removed. This is similar to the Soar Manual documentation (3.3.6.6 Right-hand side Functions) for trim function.
- register the Trim class to the list of RhsFunctionHandler
- added a test class for the trim rhs function

marinier commented:
Thank you for this! Can you replace the call to `trim()` with `strip()`? The difference is that `strip()` properly handles unicode. I don't know how well the rest of jsoar handles unicode, but at least we can make sure this part does.

rezamini committed:
- replaced trim() with strip() to ensure proper...
handling of unicodes for the RHS 'trim' function.
- added new test cases to cover scenarios where leading and trailing unicodes and whitespaces are removed from the string.

rezamini commented:
Certainly! I have made the changes and committed the updated code. In addition, I have added new test cases to verify that the updated code properly handles unicodes and whitespaces. Thank you for bringing this to my attention.

marinier commented:
Looks great! Thank you again! Merged.
